### ISSUE TITLE
Fully support of XEP-0158 (Captchas)

### ISSUE THREAD

**Issue by georgehazan** [miranda-ng/miranda-ng#173]
**Reported by HostedDinner on 17 Feb 2013 15:56 UTC**
More servers require captchas for register, so Miranda could support them ;)

The old issue on the Miranda Im Tracker:
http://code.google.com/p/miranda/issues/detail?id=1212

**Comment by georgehazan** [comment]
**Modified by watcher on 2 Jan 2014 06:29 UTC**

**Comment by georgehazan** [comment]
**Modified by watcher on 29 Jun 2014 12:05 UTC**

**Comment by LittleVulpix** [comment]
I tested registering with suchat.org, and while the captcha didn't actually "display" like an image in miranda dialogue, I did receive the captcha link in the dialogue and it was clear what to do with it (the window actually asks the user to open the captcha in a browser and type it out).

I was able to register on suchat.org and fill out the captcha without issues, confirming the registration by also being able to log in.

This can be closed. It is possible a new enhancement ticket can be opened later if someone wants to actually see the captcha image in miranda, but it's not required.

**Comment by Robyer** [comment]
@LittleVulpix So Miranda received direct link to captcha image? Then it should be simple to implement, as Miranda (and Jabber specifically) already supports showing captcha codes in dialog. As can be seen when joining captcha protected chatrooms.

**Comment by LittleVulpix** [comment]
Yes, I got a direct link to the captcha image with "if you cannot see the image, paste it in your browser" - which I did, and the image was loaded correctly in my browser, and I was able to use it to register.
